cmd/network/verification.go

```diff
@@ -464,6 +482,85 @@ func (e *EgressVerification) getSecurityGroupId(ctx context.Context) (string, er
 	return *resp.SecurityGroups[0].GroupId, nil
 }
 
+// getCABundle retrieves a cluster's CA Trust Bundle from Hive
+func (e *EgressVerification) getCABundle(ctx context.Context) (string, error) {
+	if e.cluster == nil || e.cluster.AdditionalTrustBundle() == "" {
+		// No CA Bundle to retrieve for this cluster
+		return "", nil
+	}
+
+	scheme := runtime.NewScheme()
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return "", err
+	}
+	// Register hivev1 for SelectorSyncSets
+	if err := hivev1.AddToScheme(scheme); err != nil {
+		return "", err
+	}
+
+	hive, err := utils.GetHiveCluster(e.cluster.ID())
+	if err != nil {
+		return "", err
+	}
+
+	e.log.Debug(ctx, "assembling K8s client for %s (%s)", hive.ID(), hive.Name())
+	hc, err := k8s.New(hive.ID(), client.Options{Scheme: scheme})
+	if err != nil {
+		return "", err
+	}
+
+	e.log.Debug(ctx, "searching for proxy SyncSet")
+	nsList := &corev1.NamespaceList{}
+	clusterSelector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: map[string]string{
+		"api.openshift.com/id": e.cluster.ID(),
+	}})
+	if err != nil {
+		return "", err
+	}
+
+	if err := hc.List(ctx, nsList, &client.ListOptions{LabelSelector: clusterSelector}); err != nil {
+		return "", err
+	}
+
+	if len(nsList.Items) != 1 {
```

Comment:
Extremely minor - we could save a `len` calculation/lookup here with a short statement and then reusing that in the error output.